src/net/sourceforge/kolmafia/textui/command/AsdonMartinCommand.java

```diff
@@ -48,23 +60,23 @@ private static String driveStyleName(final int index) {
     if (index < 0 || index > 8) {
       return null;
     }
-    return (String) DRIVESTYLE[index][0];
+    return DRIVESTYLE[index].name;
   }
 
   private static int currentDriveStyle() {
     List<AdventureResult> active = KoLConstants.activeEffects;
-    for (int i = 0; i < DRIVESTYLE.length; ++i) {
-      if (active.contains(DRIVESTYLE[i][2])) {
-        Integer index = (Integer) DRIVESTYLE[i][1];
-        return index.intValue();
+    for (DriveStyle driveStyle : DRIVESTYLE) {
+      if (active.contains(driveStyle.effect)) {
+        return driveStyle.driveId;
       }
     }
     return -1;
   }
 
   @Override
   public void run(final String cmd, final String parameters) {
-    if (CampgroundRequest.getCurrentWorkshedItem().getItemId() != ItemPool.ASDON_MARTIN) {
+    var workshedItem = CampgroundRequest.getCurrentWorkshedItem();
+    if (workshedItem == null || workshedItem.getItemId() != ItemPool.ASDON_MARTIN) {
```

Comment:
Just a personal style preference but I would love using parentheses so I don't have to rely on memory or a web search to confirm that workshedItem of null won't trigger a NPE.  

(workshedItem == null || (workshedItem.getItemId() != ItemPool.ASDON_MARTIN))

That makes it clearer to me :-)  Not a blocking comment, however.

Reply:
happy to, but I think there are other ways to be satisfied in this case:
* there is a test that checks null and passes
* the type of `workshedItem == null || workshedItem.getItemId()` would be `boolean || int` which isn't valid

I was expecting spotless to remove the brackets, but it didn't so I think it's fine to have them, and I've added some